protocols/identify/src/identify.rs

```diff
@@ -185,31 +269,57 @@ impl NetworkBehaviour for Identify {
             return Poll::Ready(event);
         }
 
-        if let Some(r) = self.pending_replies.pop_front() {
-            // The protocol names can be bytes, but the identify protocol except UTF-8 strings.
-            // There's not much we can do to solve this conflict except strip non-UTF-8 characters.
-            let protocols: Vec<_> = params
-                .supported_protocols()
-                .map(|p| String::from_utf8_lossy(&p).to_string())
-                .collect();
+        // Check for a pending active push to perform.
+        let peer_push = self.pending_push.iter().find_map(|peer| {
+            self.connected.get(peer).map(|conns| {
+                let observed_addr = conns
+                    .values()
+                    .next()
+                    .expect("connected peer has a connection")
+                    .clone();
+
+                let listen_addrs = listen_addrs(params);
+                let protocols = supported_protocols(params);
+
+                let info = IdentifyInfo {
+                    public_key: self.config.local_public_key.clone(),
+                    protocol_version: self.config.protocol_version.clone(),
+                    agent_version: self.config.agent_version.clone(),
+                    listen_addrs: listen_addrs.clone(),
```

Comment:
```suggestion
                    listen_addrs,
```
Clone not needed according to clippy. Seems right given `fn listen_addrs`, haven't confirmed locally.